### PR TITLE
feat: add ValidationError

### PR DESCRIPTION
**Pull request by michael-ciniawsky**
schemautilserror

### CODE REVIEW

**Comment by michael-ciniawsky** [comment]
@bebraw @d3viant0ne please review

**Comment by michael-ciniawsky** [comment]
Fixes  #5 #6 #7

File: test/index.test.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/* eslint-disable */
+import validateOptions from '../dist/validateOptions';
```
Comment (d3viant0ne) [inline test/index.test.js]: You keep defaulting to this testing approach and as I have said before, you negate most of the usefulness of using Jest.

- Jest doesn't require a build step as it's setup in defaults. babel-jest takes care of that.
- Jest runs tests async & only runs the tests related to the code you have changed, this negates that as well.
- None of the editor plugins for Jest work with this approach

Comment (michael-ciniawsky) [inline test/index.test.js]: Done

**Comment by michael-ciniawsky** [comment]
Anything else outstanding here ? Otherwise please merge and release I need this to move on with `webpack-defaults` 😛

**Comment by bebraw** [comment]
Yeah, I would merge this and get a new version out. Still a fresh package so better move fast.

**Comment by michael-ciniawsky** [comment]
It's MVP 😛 I intentionally avoided anything  AJV 'advanced'  for now and update `schema-utils` if we hit any limits with the current approach, but `Load Schema => Validate Schema => Validate Options ? true : throw new ValidationError()` works and should be fine for 90% of usecases in Loaders/Plugins, at least we can add it to all repos with the `webpack-defaults` upgrade and minor improvements follow :)

- `webpack-defaults` release with latest updates
- `schema-utils` release with this PR 

and I'm rolling again 😛 

⚠️ This is technically a breaking change, bc of `ajv@5.0.0` (no `$schema`), but I'm aware of that especially for ETWP ⚠️

**Comment by michael-ciniawsky** [comment]
@bebraw ping 😛

**Comment by d3viant0ne** [comment]
@michael-ciniawsky - Juho is still touring as far as I know so his availability is limited.

Given this is an internal tool and limited in deployment at the moment, as long as you are finished with this PR, I'll land it and publish a Major.

**Comment by michael-ciniawsky** [comment]
@d3viant0ne ohh.. sry I wasn't aware of that :)  Thx 👍 the PR is finished and tested locally with a dummy loader/plugin && webpack

validationerror

`Loader === ${Name}` in the screenshot 😛

**Comment by michael-ciniawsky** [comment]
@d3viant0ne Maybe this error format is better

```
ERROR in 'bla.js'
Module build failed: Validation Error
${Name} invalid options found: 

options.foo should be bar
```

?

**Comment by michael-ciniawsky** [comment]
:shipit: 😛

**Pull request merged**